rts/motoko-rts/src/allocator.rs

```diff
@@ -0,0 +1,43 @@
+// c.f. https://os.phil-opp.com/heap-allocation/#dynamic-memory
+
+use alloc::alloc::{GlobalAlloc, Layout};
+//use core::ptr::null_mut;
+use crate::memory::{alloc_blob, ic};
+use crate::types::Bytes;
+
+pub struct HeapAllocator;
+
+unsafe impl GlobalAlloc for HeapAllocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
```

Comment:
Unit testing is probably difficult because of the `IcMemory` injection...

Reply:
Yeah, not sure what to do about that. I was surprised that the native tests still build without warning but I guess they just use the default allocator instead. Not sure actually. Maybe I can add a lower_case test, but that wouldn't actually test this allocator....

Reply:
I could try, but probably it is not easy.
Btw: This allocator is not used in the RTS unit tests because this module is excluded for test compilation (in `lib.rs`).

Reply:
I wouldn't worry about it. We have the .mo tests anyway.